pkgs/by-name/so/sobjectizer/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  callPackage,
+  buildExamples ? false,
+  buildTests ? false,
+  buildAll ? false,
+}:
+
+# Derivation build example:
+#   $ nix-build -A sobjectizer
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sobjectizer";
+  version = "5.8.4";
+
+  src = fetchFromGitHub {
+    owner = "Stiffstream";
+    repo = "sobjectizer";
+    rev = "48240cff83d758f0e7fc413ec408536147a374ed";
+    hash = "sha256-tIqWgd6TppHfqZk3XHzhG0t+Nn8BQCTP81UD7ls67UE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeDir = "../dev";
+
+  cmakeFlags = lib.optionals (buildExamples) [ "-DBUILD_EXAMPLES=ON" ] ++
+               lib.optionals (buildTests)    [ "-DBUILD_TESTS=ON" ] ++
```

Comment:
Perhaps `finalAttrs.doCheck`?

Reply:
In purpose of learning:
Why I should use `finalAttrs.doCheck` vs `doCheck`?

Thank you!

Reply:
`finalAttrs.doCheck` references the `doCheck` value set in the file, which helps you not have to repeat yourself.
It also respects `override` directives that are external to the file, so that one can disable the tests without having to change the file's contents.